TODO.md

```diff
@@ -0,0 +1,26 @@
+## TODO
+
+* Add datapoints in provisioning logic for Influx
+* Be more specific in JSON Schema
+* Load things from NextToken when retreiving spot price history
+* Overwrite UserData with temporary taskcluster credentials
+  * Include the API endpoint for the slave to hit the hello
+    endpoint
+  * include the instanceId scope!
+* Encrypt UserData using opengpg.js
+* Send pulse messages for
+  * killing an instance
+  * cancelling a spot request
+  * creating a spot request
+* Use Joi/assert to better control the input into functions
+* Verify that errors in provisioning don't kill the entire
+  provisioner
+* Kill capacity which exceeds maxCapacity
```

Comment:
This is blocking. If we're not killing nodes if we launch too many.. (or too many just shows up for no reason). We have problem. We should probably log cases where we kill nodes because we're over the total max. It'll happen occasionally because ec2 is eventually consistent, but it shouldn't happen often.

We need it as a last guard against bad things (tm).